domains/client/domain-operator/src/fraud_proof.rs

```diff
@@ -389,14 +389,20 @@ where
             &self.storage_key_provider,
         )?;
 
-        let invalid_inherent_extrinsic_proof = InvalidInherentExtrinsicProof::generate(
+        let domain_runtime_upgraded_proof = DomainRuntimeUpgradedProof::generate(
             &self.storage_key_provider,
             self.consensus_client.as_ref(),
-            domain_id,
             consensus_block_hash,
             maybe_runtime_id,
```

Comment:
We can also replace `BlockDigest` with `DomainRuntimeUpgrades` to determine if runtime upgrades happen in `is_domain_runtime_upgraded_at`

Reply:
I don't know how to do this, sorry! For the next few months, can provide more details about how to do things in tickets and reviews?

You can see what I tried in commit 6077082fc06ceb8c16e4704ea7e60ac19000a2f5 of https://github.com/autonomys/subspace/tree/fraud-proof-data-digest-upgraded-at-via-proof

I couldn't get the value of `DomainRuntimeUpgrades` in `is_domain_runtime_upgraded_at()`, because there is no runtime API for it. So I tried creating and verifying a storage proof to get that value, but I think that didn't work because the proof hadn't been submitted yet.

Do you want me to add a new runtime API?

Reply:
>Do you want me to add a new runtime API?

Yeah, that is what I mean, sorry for not being clear, let's leave this to the next PR.